### ISSUE TITLE
LayerName is not supporting intrinsic functions in AWS::Serverless::LayerVersion?

### ISSUE THREAD

**Issue by pshvets** [awslabs/serverless-application-model#754]
**Description:**

For some reason !Sub stopped working in LayerName sometimes back.

**Steps to reproduce the issue:**
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Parameters:

  NameSpace:
    Type: String
```
```yaml
CommonLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      LayerName: !Sub '${NameSpace}-AuroraCommonLayer'
      Description: Common Layer
      ContentUri: ../artifacts/python/CommonLayer.zip
      CompatibleRuntimes:
        - python3.7
```

**Observed result:**
`Failed to create the changeset: Waiter ChangeSetCreateComplete failed: Waiter encountered a terminal failure state Status: FAILED. Reason: Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [CommonLayer] is invalid. Could not resolve parameter for 'LayerName' or parameter is not a String.
`
**Expected result:**
Should (!! and used to !!) work

**Comment by pshvets** [comment]
It seems like it was fixed in #753 
Could you please confirm?

**Comment by keetonian** [comment]
This was unintended; we rolled back the deployment and submitted a fix for this issue. Thank you for reporting, it is indeed fixed.
